_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,416 @@
+
+Throttling might cause the reassignment to take longer to complete.
+
+1. If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+2. If the throttle is too high, the overall health of the cluster may be impacted.
+
+## Actions that can be executed while using the tool
+
+It has three different actions:
+
+| Action | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file. This actions is optional if we already have `reassignment.json` file with us.|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `--generate` command or written by the user itself. |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any replication quotas (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. |
+
+## Example time
+
+Let us understand the working of this tool with an interesting problem.
+Suppose we have 5 Kafka Brokers and after looking at the partition details we realize that the brokers are not too busy, so we can scale them down to 3.
+Through this example we will take a look at how the three actions of the Kafka reassignment partition tool(`--generate`, `--execute` and `--verify`) works.
+We will generate the JSON data that will be used in the `reassignment.json` file.
+We will then assign the partitions to the remaining broker using the `reassignment.json` file.
+The Kafka Cluster will be configured to use PLAIN listener.
+
+Before proceeding towards the steps. Let's discuss one more curious question. Can you scale down any pod you want through this process?
+So the answer to this question is no. Wondering why?
+It is due to the fact Strimzi uses StatefulSets to manage broker pods. So you cannot remove any pod from the cluster.
+You can only remove one or more of the highest numbered pods from the cluster. For example, in a cluster of 5 brokers the pods are named `<CLUSTER-NAME>-kafka-0` up to `<CLUSTER-NAME>-kafka-4`.
+If you decide to scale down by two brokers, then `<CLUSTER-NAME>-kafka-4` and `<CLUSTER-NAME>-kafka-3` will be removed.
+
+### Preparing to scale down the number of Kafka Brokers
+
+This example will use a Kafka cluster deployed with the Strimzi Cluster Operator.
+Before we start with anything, we have to install the Strimzi Cluster Operator and deploy the Kafka cluster.
+You can install the Cluster Operator with any installation method you prefer.
+The Kafka cluster is then deployed with the PLAIN listener enabled on port 9092.
```

Comment:
Some readers might want to follow along and be prepared to stand up a cluster to do so (and for experimenting), others might have an existing cluster they actually want to scale down, so you need to provide enough context for any reader to be able to apply what they're reading to their situation.

I think you need to say something like "If you want to follow along you'll need a cluster that looks like...", and also something like "If you already have an existing cluster with a plain listener you can skip this section"